heareval/predictions/task_predictions.py

```diff
@@ -216,44 +386,32 @@ def create_events_from_prediction(
     # Convert probabilities to binary vectors based on threshold
     predictions = (predictions > threshold).astype(np.int8)
 
-    # Difference between each timestamp to find event boundaries
-    pred_diff = np.diff(predictions, axis=0)
-
-    # This is slow, but for each class look through all the timestamp predictions
-    # and construct a set of events with onset and offset timestamps.
     event_tree = IntervalTree()
-    for class_idx in range(predictions.shape[1]):
-        # Check to see if this starts with an active event
-        current_event = []
-        if predictions[0][class_idx] == 1:
-            assert pred_diff[0][class_idx] != 1
-            current_event = [timestamps[0]]
-
-        for t in range(pred_diff.shape[0]):
-            # New onset
-            if pred_diff[t][class_idx] == 1:
-                assert len(current_event) == 0
-                current_event = [timestamps[t]]
-
-            # Offset for current event
-            elif pred_diff[t][class_idx] == -1:
-                assert len(current_event) == 1
-                current_event.append(timestamps[t + 1])
-                event_duration = current_event[1] - current_event[0]
-
-                # Add event if greater than the minimum duration threshold
-                if event_duration >= min_duration:
-                    event_tree.addi(
-                        begin=current_event[0], end=current_event[1], data=class_idx
-                    )
-                current_event = []
-
+    for label in range(predictions.shape[1]):
+        for group in more_itertools.consecutive_groups(
+            np.where(predictions[:, label])[0]
+        ):
+            grouptuple = tuple(group)
+            startidx, endidx = (grouptuple[0], grouptuple[-1])
+
+            start = timestamps[startidx]
+            end = timestamps[endidx]
+            # Add event if greater than the minimum duration threshold
+            if end - start >= min_duration:
+                # We probably don't need this interval tree and can remove it maybe
+                # from requirements
+                event_tree.addi(begin=start, end=end, data=label)
+
+    # Er why return an intervaltree when we immediately postprocess it to a dict?
```

Comment:
I still want to remove this intervaltree nonsense, it doesn't add anything as far as I can tell?
I don't think we need it at all

Reply:
One of the remaining things I think we get from intervaltree is the sort of all events by the start timestamp -- so that is kind of nice. I'm curious whether that is necassary for the sed eval stuff. Either way, that wouldn't be too challenging to implement if it is requirement and we have everything we need it looks like to go straight from the consecutive groups to a dictionary w/o the intervaltree

Reply:
So if you wanna ditch the interval tree I'd say go for it